### PR TITLE
Issue #6288: In user_register_form(), use the value of $admin instead of calling again user_access('administer users')

### LINKED ISSUE #6288: In user_register_form(), use the value of $admin instead of calling again user_access('administer users')
The first two lines of [`user_register_form()`](https://docs.backdropcms.org/api/backdrop/core%21modules%21user%21user.module/function/user_register_form/1) are the following ones.

```php
  global $user;
  $admin = user_access('administer users');
```

Later, that function uses the following code.

```php
  elseif (user_access('administer users')) {
    $path = 'admin/people';
  }
```

Instead of calling again `user_access('administer users')`, the code should check the value of `$admin`.

```php
  elseif ($admin) {
    $path = 'admin/people';
  }
```

That is exactly what the code some lines before that does.

```php
  if ($admin) {
    // Redirect back to page which initiated the create request;
    // usually admin/people/create.
    $form_state['redirect'] = $_GET['q'];
  }
```

### PR DESCRIPTION
**Pull request by avpaderno**
Fixes https://github.com/backdrop/backdrop-issues/issues/6288

### CODE REVIEW

File: core/modules/user/user.module [hunk 1]
```diff
@@ -1625,7 +1625,8 @@ function user_menu_alter(&$items) {
  */
 function user_admin_bar_output_alter(&$content) {
   // Admin menu shows both the parent menu item and its children.
-  // Remove the duplicate child element so "Manage fields" doesn't show up twice.
+  // Remove the duplicate child element, so "Manage fields" doesn't show up
```
Comment (bugfolder) [inline core/modules/user/user.module]:
```suggestion
  // Remove the duplicate child element so "Manage fields" doesn't show up
```
The added comma is ungrammatical.

Comment (avpaderno) [reply core/modules/user/user.module]: I guess it depends from whether the last clause is considered independent or dependent. Probably it is a dependent clause as the clause in the following sentence.

> I went to the store so I could buy tomatoes.

**Comment by avpaderno** [comment]
(The failing tests are upgrade tests which keep to randomly fail.)

**Comment by bugfolder** [comment]
`testLayoutUpgrade` is notorious for randomly failing, unrelated to the PR. While it's nice to get all 12 tests to pass, if the failing tests are unrelated and due to the test gremlin, we could still proceed. So the issue just needs human testing (which should be easy) and code review (ditto, since the change is small and simple). I'll try to get time a little later to do this.

**Comment by avpaderno** [comment]
Perfect! Now, it is just CSpell complaining about unknown words.

**Review by klonos — APPROVED** [review]

**Comment by quicksketch** [comment]
Thanks @kiamlaluno! Merged into 1.x and 1.26.x.

**Pull request merged**